ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/add/AbstractAddPartitionAnalyzer.java

```diff
@@ -70,7 +70,7 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
     }
 
     AlterTableAddPartitionDesc desc = new AlterTableAddPartitionDesc(table.getDbName(), table.getTableName(),
-        ifNotExists, partitions);
+        ifNotExists, partitions, conf.get("hcat.dynamic.partitioning.custom.pattern", null));
```

Comment:
why do you need to specify `null` default and why not use. constant `HCAT_DYNAMIC_CUSTOM_PATTERN`?